### DISCUSSION TITLE
Psalm with PHP-DI

### DISCUSSION THREAD

**Discussion by gillesdalle** [vimeo/psalm#6285] [Q&A] [answered]
My snippet :  [https://psalm.dev/r/bd1da8091e](https://psalm.dev/r/bd1da8091e)

I use PHP-DI to define injections using annotations ([https://php-di.org/doc/annotations.html](https://php-di.org/doc/annotations.html)).

The MissingConstructor error is a false positive. Is there a way to make psalm understand this annotation?

Thanks

**Comment by orklah** [comment, marked as answer]
It could be handled through a plugin hook without too much trouble. Here's an example of how `@required` was handled on methods in the Symfony Psalm plugin:
https://github.com/psalm/psalm-plugin-symfony/pull/100/files

A PHP-DI plugin does not yet exists, it could be a good thing to have :)
